src/181810.c

```diff
@@ -13,21 +13,101 @@ BSS ActorPart* gSpeakingActorPart;
 
 u8* gBattleDmaDest = NULL;
 
-u8 D_80284004[] = { 59, 60, 61, 62, 63, 64, 65, 66, 67, 255 };
-u8 D_80284010[] = { 126, 127, 128, 129, 130, 131, 132, 133, 134, 255 };
-u8 D_8028401C[] = { 189, 190, 255 };
-u8 D_80284020[] = { 170, 171, 255 };
-u8 D_80284024[] = { 195, 197, 255 };
-u8 D_80284028[] = { 77, 79, 78, 80, 255 };
-u8 D_80284030[] = { 81, 82, 255};
-u8 D_80284034[] = { 83, 84, 255 };
-u8 D_80284038[] = { 87, 88, 255 };
-u8 D_8028403C[] = { 85, 86, 255 };
-u8 D_80284040[] = { 89, 90, 255 };
-
-u8* D_80284044[] = {
-    D_80284004, D_80284010, D_8028401C, D_80284020, D_80284024, D_80284028, D_80284030, D_80284034, D_80284038,
-    D_8028403C, D_80284040, NULL,
+u8 ActorTypesGhost[] = {
+    ACTOR_TYPE_DUPLIGHOST,
+    ACTOR_TYPE_GHOST_GOOMBARIO,
+    ACTOR_TYPE_GHOST_KOOPER,
+    ACTOR_TYPE_GHOST_BOMBETTE,
+    ACTOR_TYPE_GHOST_PARAKARRY,
+    ACTOR_TYPE_GHOST_BOW,
+    ACTOR_TYPE_GHOST_WATT,
+    ACTOR_TYPE_GHOST_SUSHIE,
+    ACTOR_TYPE_GHOST_LAKILESTER,
+    -1
```

Comment:
Given this is a u8 list it'd be better to use 0xFF or perhaps define an ACTOR_TYPE_LIST_END and use it in the below funcs too

Reply:
I disagree on this. Many different list types in the game use -1 to indicate "end of list", regardless of storage size. I don't think its a good idea to proliferate END_OF_X for each type. The only one I've been using is PTR_LIST_END for (void*)-1. Unfortunately, defining a generic LIST_END as -1 is not possible, as 0 is sometimes used for end of list as well.

Reply:
yeah I forgot to change the checks to use -1 in the function. I prefer -1 to 0xFF because 0xFF feels like just a large number, where -1 to me indicates a non-normal value. but I'll change everything back to 0xFF for now

Reply:
I agree with using -1 but I'd assume that using that in the function might produce different asm? If not then yeah lets use -1 everywhere :)

Reply:
it dont work :despair: